forge-game/src/main/java/forge/game/player/Player.java

```diff
@@ -3242,12 +3243,9 @@ public void createCommanderEffect() {
         String moved = "Event$ Moved | ValidCard$ " + validCommander + " | Secondary$ True | Optional$ True | OptionalDecider$ You | CommanderMoveReplacement$ True ";
         if (game.getRules().hasAppliedVariant(GameType.TinyLeaders)) {
             moved += " | Destination$ Graveyard,Exile | Description$ If a commander would be put into its owner's graveyard or exile from anywhere, that player may put it into the command zone instead.";
-        }
-        else if (game.getRules().hasAppliedVariant(GameType.Oathbreaker)) {
-            moved += " | Destination$ Graveyard,Exile,Hand,Library | Description$ If a commander would be exiled or put into hand, graveyard, or library from anywhere, that player may put it into the command zone instead.";
         } else {
-            // rule 903.9b
-            moved += " | Destination$ Hand,Library | Description$ If a commander would be put into its owner's hand or library from anywhere, its owner may put it into the command zone instead.";
+            // Rule CR 903.9
```

Comment:
bad change, that's not how it works

Reply:
"903.9. A commander may return to the command zone during a Commander game.

903.9a If a commander is in a graveyard or in exile and that object was put into that zone since the last time state-based actions were checked, its owner may put it into the command zone. This is a state-based action. See rule 704.

903.9b If a commander would be put into its owner’s hand or library from anywhere, its owner may put it into the command zone instead. This replacement effect may apply more than once to the same event. This is an exception to rule 614.5."

Unless I'm mistaken and the 903.9a part is covered elsewhere (which i couldn't find in code, nor could i get to work in testing) this change is the best approximation of how this should work.

Reply:
That said, I can't seem to get the 903.9a effect to trigger, trying Kroxa as a commander and if he gets exiled by say Kalitas or Valgavoth, no prompt to put him back in the command zone happens. Investigating

Reply:
Found the point at which to add the variant check for DuelCommander that was missing, thanks for catching that